blazor/datagrid/searching.md

```diff
@@ -283,40 +598,289 @@ By default, the datagrid will initiate searching operation after the Enter key i
+                    Orders.Add(new OrderData(10254, "CHOPS", "9","Chop-suey Chines", "Switzerland", "Chop-suey Chinese"));
+                    Orders.Add(new OrderData(10255, "RICSU", "5","Bern", "Switzerland", "Richter Supermarkt"));
+                    Orders.Add(new OrderData(10256, "WELLI", "3","Genève", "Brazil", "Wellington Importadora"));
+                    code += 5;
+                }
+            }
+            return Orders;
+        }
 
+        public int OrderID { get; set; }
+        public string CustomerID { get; set; }
+        public string EmployeeID { get; set; }
+        public string ShipCity { get; set; }
+        public string ShipCountry { get; set; }
+        public string ShipName { get; set; }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+{% previewsample "https://blazorplayground.syncfusion.com/embed/BDrgXFLzhbsDRpHF?appbar=false&editor=false&result=true&errorlist=false&theme=bootstrap5" %}
+
+
+## Perform search operation in Grid using multiple keywords
+
+In addition to searching with a single keyword, the Grid component offers the capability to perform a search operation using multiple keywords. This feature enables you to narrow down your search results by simultaneously matching multiple keywords. It can be particularly useful when you need to find records that meet multiple search conditions simultaneously.
+
+The following example demonstrates, how to perform a search with multiple keywords in the grid by using the ```query``` property .
+
+{% tabs %}
+{% highlight razor tabtitle="Index.razor" %}
+@using Syncfusion.Blazor.Grids
+@using Syncfusion.Blazor.Inputs
+@using Syncfusion.Blazor.Buttons
+@using Syncfusion.Blazor.Data
+@using BlazorApp1.Data
+
+<SfTextBox Input="OnInput"></SfTextBox>
+<SfRadioButton TChecked="bool" ValueChange="DariRadioButton" Label="Lucas"></SfRadioButton>
+
+<SfGrid @ref="DefaultGrid" DataSource="@Orders" Query="@SearchQuery">
+    <GridColumns>
+        <GridColumn Field=@nameof(OrderData.OrderID) HeaderText="Order ID" Width="120"></GridColumn>
+        <GridColumn Field=@nameof(OrderData.CustomerID) HeaderText="Customer Name" Width="150"></GridColumn>
+        <GridColumn Field=@nameof(OrderData.Paid) Width="150"></GridColumn>
+        <GridColumn Field=@nameof(OrderData.OrderDate) HeaderText=" Order Date" Format="d" Type="ColumnType.Date" TextAlign="TextAlign.Right" Width="130"></GridColumn>
+        <GridColumn Field=@nameof(OrderData.Freight) HeaderText="Freight" Format="C2" TextAlign="TextAlign.Right" Width="120"></GridColumn>
+    </GridColumns>
+</SfGrid>
+
+@code {
+    public Query SearchQuery { get; set; }
+    private SfGrid<OrderData> DefaultGrid;
+    public List<OrderData> Orders { get; set; }
+    WhereFilter ColumnPredicate = new WhereFilter();
+    List<WhereFilter> Predicate = new List<WhereFilter>();
     public void OnInput(InputEventArgs args)
     {
-        this.DefaultGrid.Search(args.Value);
+        Predicate = new List<WhereFilter>();
```

Comment:
need to modify this sample. behaving wrongly when radio button is selected first

Reply:
which is already used in old UG.